### PR TITLE
exercises done 

### PR DESCRIPTION
**Pull request by qKarpej**
sorry, I forgot to do pull request

### CODE REVIEW

File: exercises/01_sort4/sort4.go [hunk 1]
```diff
@@ -3,6 +3,10 @@ package main
 import "fmt"
 
 func minmax(a, b int) (mn, mx int) {
+
```
Comment (yarcat) [inline exercises/01_sort4/sort4.go]: nit: try to avoid unnecessary blank lines

File: exercises/02_sort3/sort3.go [hunk 2]
```diff
@@ -2,7 +2,28 @@ package main
 
 import "fmt"
 
+func minmax(a, b int) (mn, mx int) {
+	return min(a, b), max(a, b)
```
Comment (yarcat) [inline exercises/02_sort3/sort3.go]: nit: this is the place, where you don't want to return `min(a, b), max(a, b)`:
```go
func minmax(a, b int) (mn, mx int) {
  if a < b {
    return a, b
  }
  return b, a
}
```
we'd expected that solution. it's more efficient.

File: exercises/02_sort3/sort3_test.go [hunk 3]
```diff
@@ -2,7 +2,7 @@ package main
 
 import "testing"
 
-func TestSort4(t *testing.T) {
+func TestSort3(t *testing.T) {
```
Comment (yarcat) [inline exercises/02_sort3/sort3_test.go:5]: I'll give you back the point lost in `minmax` for fixing the test name!

File: exercises/03_reverse/reverse.go [hunk 4]
```diff
@@ -1,4 +1,13 @@
...
+func reverse(n int) int {
+
+	r := 0
+	for ; n != 0; n /= 10 {
+		r = n%10 + (r * 10)
+	}
+	return r
+}
 func main() {
+
```
Comment (yarcat) [inline exercises/03_reverse/reverse.go]: Why didn't you write the program??? It's a part of the assignment

File: exercises/04_power/power_test.go [hunk 5]
```diff
@@ -5,11 +5,21 @@ import "testing"
 // TestPow tests a Pow(x, y) function, which returns integer x**y.
 func TestPow(t *testing.T) {
 	for _, tc := range []struct {
-		x, y uint
-		want uint
+		x, y int
+		want int
 	}{
-		// Test cases.
+		{x: 1, y: 2, want: 1},
```
Comment (yarcat) [inline exercises/04_power/power_test.go:11]: You program wouldn't work correctly for the following missing test scenarios:
```
{x: 0, y: 0, want: 1},
{x: 3, y: 0, want: 1},

File: exercises/04_power/power_test.go [hunk 6]
```diff
@@ -5,11 +5,21 @@ import "testing"
 // TestPow tests a Pow(x, y) function, which returns integer x**y.
 func TestPow(t *testing.T) {
 	for _, tc := range []struct {
-		x, y uint
-		want uint
+		x, y int
```
Comment (yarcat) [inline exercises/04_power/power_test.go]: Please note that `uint` was specified here on purpose - now you miss negative tests!

File: exercises/04_power/power_test.go [hunk 7]
```diff
@@ -5,11 +5,21 @@ import "testing"
...
+		want int
 	}{
-		// Test cases.
+		{x: 1, y: 2, want: 1},
+		{x: 2, y: 2, want: 4},
+		{x: 3, y: 2, want: 9},
+		{x: 4, y: 2, want: 16},
+		{x: 5, y: 2, want: 25},
+		{x: 6, y: 2, want: 36},
```
Comment (yarcat) [inline exercises/04_power/power_test.go]: You have too many tests with the same `y`. It's always a good idea to think about the tests this way:
1. Possibly check all corner cases. In this case these are `(0, 0), (0, 1), (1, 0), (1, 1), (x, 0), (0, x)` (where `x` is any number such as `x > 1` e.g. `17`)
2. This about common cases, and check few most representative. In this case these would be `(x, y)`, where `x > 1 && y > 1` e.g. `(13, 2), (2, 13), (10, 10)`.
That's enough

File: exercises/04_power/power.go [hunk 8]
```diff
@@ -2,19 +2,21 @@ package main
 
 import "fmt"
 
-func Pow(x, y int) int {
+func Pow(x, y uint) uint {
+	if y == 0 {
+		x = 1
+		return x
```
Comment (zayac) [inline exercises/04_power/power.go:8]: you may just write `return 1`
